apps/challenges/views.py

```diff
@@ -3354,3 +3354,103 @@ def pwc_task_dataset(request):
             "error": "You are not authorized to make this request. Please ask EvalAI admin to add you as a staff user for accessing this API."
         }
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
+
+
+@api_view(["GET", "PATCH", "DELETE"])
+@throttle_classes([UserRateThrottle])
+@permission_classes(
+    (
+        permissions.IsAuthenticatedOrReadOnly,
+        HasVerifiedEmail,
+        IsChallengeCreator,
+    )
+)
+@authentication_classes((JWTAuthentication, ExpiringTokenAuthentication))
+def update_allowed_email_ids(request, challenge_pk, phase_pk):
+    """
+    API to GET/UPDATE/DELETE email ids from challenge phase allowed email ids
+    Arguments:
+        challenge_pk {int} -- Challenge primary key
+        phase_pk {int} -- Challenge phase primary key
+    Returns:
+        {dict} -- Response object
+    """
+    challenge = get_challenge_model(challenge_pk)
+
+    challenge_phase = get_challenge_phase_model(phase_pk)
+    allowed_email_ids = challenge_phase.allowed_email_ids
+
+    if allowed_email_ids is None:
+        allowed_email_ids = []
+
+    if request.method != "GET":
+        if request.data.get("allowed_email_ids") is not None:
+            if not isinstance(request.data["allowed_email_ids"], list):
+                response_data = {
+                    "error": "Field allowed_email_ids should be a list."
+                }
+                return Response(
+                    response_data, status=status.HTTP_400_BAD_REQUEST
+                )
+        else:
+            response_data = {"error": "Field allowed_email_ids is missing."}
+            return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+    if request.method == "GET":
+        serializer = ChallengePhaseCreateSerializer(
+            challenge_phase, context={"request": request}
+        )
+        response_data = {
+            "challenge_phase_pk": phase_pk,
```

Comment:
We don't need this `phase_pk` here as the challenge host already knows it while calling the API?